packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -175,8 +175,20 @@ class FlutterDevice {
     if (flutterViews == null || flutterViews.isEmpty)
       return;
     final List<Future<void>> futures = <Future<void>>[];
+    // If any of the flutter views are paused, we might not be able to
```

Comment:
Why is this only a problem for `flutterExit`. Wouldn't it also cause issues with other service protocol extensions, as well?

Reply:
Some of the methods specifically catch or handle this case, like:

https://github.com/flutter/flutter/blob/e3ee5c6bbb80732c30e18cde26602f9f7f018aa2/packages/flutter_tools/lib/src/run_hot.dart#L704

Some don't. I can fix these separately, but we'll need this fix too so that we don't leave these apps open.

Reply:
Filled https://github.com/flutter/flutter/issues/36214 to track